dali/test/python/test_operator_readers_numpy.py

```diff
@@ -75,163 +87,108 @@ def NumpyReaderPipeline(path, batch_size, device="cpu", file_list=None, files=No
-
-
-def test_batch_files():
-    with tempfile.TemporaryDirectory(prefix = gds_data_root) as test_data_root:
-        # create files
-        num_samples = 20
-        batch_size = 4
-        filenames = []
-        arr_np_list = []
-        for index in range(0,num_samples):
-            filename = os.path.join(test_data_root, "test_{:02d}.npy".format(index))
-            filenames.append(filename)
-            create_numpy_file(filename, (5, 2, 8), np.float32, False)
-            arr_np_list.append(np.load(filename))
-        arr_np_all = np.stack(arr_np_list, axis=0)
-
-        for device in ["cpu", "gpu"] if is_gds_supported() else ["cpu"]:
-            yield check_batch, None, batch_size, num_samples, device , arr_np_all, None, filenames
-
-def test_batch_files_cache_headers():
-    with tempfile.TemporaryDirectory(prefix = gds_data_root) as test_data_root:
-        # create files
-        num_samples = 20
-        batch_size = 4
-        filenames = []
-        arr_np_list = []
-        for index in range(0,num_samples):
-            filename = os.path.join(test_data_root, "test_{:02d}.npy".format(index))
-            filenames.append(filename)
-            create_numpy_file(filename, (5, 2, 8), np.float32, False)
-            arr_np_list.append(np.load(filename))
-        arr_np_all = np.stack(arr_np_list, axis=0)
-
-        for device in ["cpu", "gpu"] if is_gds_supported() else ["cpu"]:
-            yield check_batch, None, batch_size, num_samples, device , arr_np_all, None, filenames, True
-
-
-def test_batch_files_arg():
-    with tempfile.TemporaryDirectory(prefix = gds_data_root) as test_data_root:
-        # create files
-        num_samples = 20
-        batch_size = 4
-        filenames = []
-        arr_np_list = []
-        for index in range(0,num_samples):
-            rel_name = "test_{:02d}.npy".format(index)
-            filename = os.path.join(test_data_root, rel_name)
-            filenames.append(rel_name)
-            create_numpy_file(filename, (5, 2, 8), np.float32, False)
-            arr_np_list.append(np.load(filename))
-        arr_np_all = np.stack(arr_np_list, axis=0)
-
-        # create pipe
-        for num_threads in [1, 2, 4, 8]:
-            pipe = NumpyReaderPipeline(path = test_data_root,
-                                       files = filenames,
-                                       batch_size = batch_size,
-                                       num_threads = num_threads,
-                                       device_id = 0)
-            pipe.build()
-
-            for batch in range(0, num_samples, batch_size):
-                pipe_out = pipe.run()
-                arr_rd = pipe_out[0].as_array()
-                arr_np = arr_np_all[batch:batch+batch_size, ...]
+def test_types_and_shapes():
+    cache_header_information = False
+    for device in ["cpu", "gpu"] if is_gds_supported() else ["cpu"]:
+        for fortran_order in [False, True, None]:
```

Comment:
Maybe we can just go with None that means random, what do you think?

Reply:
You mean, only test random?

Reply:
Yes, only test random. Food for thoughts.